hikari/impl/rest.py

```diff
@@ -4377,3 +4392,88 @@ def fetch_scheduled_event_users(
         return special_endpoints_impl.ScheduledEventUserIterator(
             self._entity_factory, self._request, newest_first, str(start_at), guild, event
         )
+
+    async def fetch_skus(
+        self, application: snowflakes.SnowflakeishOr[guilds.PartialApplication]
+    ) -> typing.Sequence[monetization.SKU]:
+        response = await self._request(routes.GET_APPLICATION_SKUS.compile(application=str(int(application))))
+        assert isinstance(response, list)
+
+        return [self._entity_factory.deserialize_sku(payload) for payload in response]
+
+    async def fetch_entitlements(
+        self,
+        application: snowflakes.SnowflakeishOr[guilds.PartialApplication],
+        /,
+        *,
+        user: undefined.UndefinedOr[snowflakes.SnowflakeishOr[users.PartialUser]] = undefined.UNDEFINED,
+        guild: undefined.UndefinedOr[snowflakes.SnowflakeishOr[guilds.PartialGuild]] = undefined.UNDEFINED,
+        before: undefined.UndefinedOr[snowflakes.SearchableSnowflakeish] = undefined.UNDEFINED,
+        after: undefined.UndefinedOr[snowflakes.SearchableSnowflakeish] = undefined.UNDEFINED,
+        limit: undefined.UndefinedOr[int] = undefined.UNDEFINED,
+        exclude_ended: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+    ) -> typing.Sequence[monetization.Entitlement]:
+        query = data_binding.StringMapBuilder()
+        if user is not undefined.UNDEFINED:
+            query.put("user_id", str(int(user)))
+
+        if guild is not undefined.UNDEFINED:
+            query.put("guild_id", str(int(guild)))
+
+        if before is not undefined.UNDEFINED:
+            if isinstance(before, datetime.datetime):
+                query.put("before", str(snowflakes.Snowflake.from_datetime(before)))
+            else:
+                query.put("before", str(int(before)))
+
+        if after is not undefined.UNDEFINED:
+            if isinstance(after, datetime.datetime):
+                query.put("after", str(snowflakes.Snowflake.from_datetime(after)))
+            else:
+                query.put("after", str(int(after)))
+
+        if limit is not undefined.UNDEFINED:
+            query.put("limit", limit)
+
+        if exclude_ended is not undefined.UNDEFINED:
+            query.put("exclude_ended", exclude_ended)
+
+        response = await self._request(
+            routes.GET_APPLICATION_ENTITLEMENTS.compile(application=str(int(application))), query=query
+        )
```

Comment:
```suggestion
        if undefined.count(before, after) < 1:
            raise TypeError("Expected no kwargs, or a maximum of one of 'before', 'after'")
       
        query = data_binding.StringMapBuilder()
        query.put("user_id", user)
        query.put("guild_id", guild)
        query.put("limit", limit)
        query.put("exclude_ended", exclude_ended)

        if before is not undefined.UNDEFINED:
            if isinstance(before, datetime.datetime):
                query.put("before", snowflakes.Snowflake.from_datetime(before)))
            else:
                query.put("before", str(int(before)))

        if after is not undefined.UNDEFINED:
            if isinstance(after, datetime.datetime):
                query.put("after", str(snowflakes.Snowflake.from_datetime(after)))
            else:
                query.put("after", str(int(after)))

        route = routes.GET_APPLICATION_ENTITLEMENTS.compile(application=application)
        response = await self._request(route, query=query)
```

And the logic for before and after should probably be extracted and just passed as a converter when passed to `query.put`, as it seems to be used a lot in this same module